src/nmodl/parse1.ypp

```diff
@@ -1050,14 +1050,8 @@ nrnlist: NAME
 	;
 valence: /*nothing*/
 		{$$ = ITEM0;}
-	| VALENCE real
+	| VALENCE number
```

Comment:
Can we keep this as a separate issue? My limited chemistry understanding is that VALENCE must be a small, positive integer. Even though it should be a non-change, it's not related to the bug we're fixing :) For those reasons, I'd suggest a separate issue which can be discussed as a separate topic.

Reply:
Perhaps beside the point, but my long ago choice of VALENCE, instead of CHARGE, I consider to be a mistake that, because of long usage, we are now stuck with. However, for those who care, in src/nmodl/init.cpp, there is the "synonym"
```
~/neuron/nrn/src/nmodl$ grep -r CHARGE
init.cpp:                {"CHARGE", VALENCE},
```
Anyway, this fix looks good to me. Putting it in a separate issue is a good idea.